lib/acl.c

```diff
@@ -45,12 +45,31 @@ enum {
 
 /* Callback function for when there's no classification match. */
 static inline int
-drop_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
-	__attribute__((unused)) struct gatekeeper_if *iface)
+match_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct gatekeeper_if *iface)
 {
 	unsigned int i;
-	for (i = 0; i < num_pkts; i++)
-		rte_pktmbuf_free(pkts[i]);
+	for (i = 0; i < num_pkts; i++) {
+		unsigned int j;
+		bool free_pkt = false;
+		/* Start from the first one. */
+		for (j = 1; j < iface->acl_func_count; j++) {
+			int ret = iface->ext_funcs[j](pkts[i], iface);
+			if (ret == -ENOENT)
+				free_pkt = true;
+			else {
```

Comment:
Good suggestion! Since we will classify the unmached packets using the matching functions at the step that packets are rearranged in the matrix ``pkts``. After that, the packets still with type ``RTE_ACL_INVALID_USERDATA`` should be dropped. So, I added function ``drop_unmatched_ipv6_pkts`` for type ``RTE_ACL_INVALID_USERDATA``.